curriculum/challenges/english/blocks/lab-pricing-plans-layout/68e45a2b4ab9b2f48b1cc6df.md

```diff
@@ -0,0 +1,272 @@
+---
+id: 68e45a2b4ab9b2f48b1cc6df
+title: Design a Pricing Plans Layout Page
+challengeType: 25
+dashedName: design-a-pricing-plans-layout-page
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories**
+
+1. Your page should have an `h1` element with the text `Pricing Plans`.
+2. Your page should have three `div` elements with the class `pricing-card` to represent the pricing plans.
+   - One of your `div` elements should have the class `pro-plan` in addition to the `pricing-card` class.
+   - One of your `div` elements should have the class `basic-plan` in addition to the `pricing-card` class.
+   - One of your `div` elements should have the class `premium-plan` in addition to the `pricing-card` class.
+3. Each of your `.pricing-card` elements should use Flexbox to stack its content in a column and justify the space between the children using `space-between`. Each element should also set the `flex` property to `0 0 200px` to give it a consistent width and prevent it from growing or shrinking in the layout.
+4. The `basic-plan` element should appear first in the layout. You should use the `order` property for this.
+5. The `pro-plan` element should appear second in the layout. You should use the `order` property and set its `flex-grow` property to `2` so it takes up more space than the other plans.
+6. The `premium-plan` element should come last in the layout. You should use the `order` property for this.
+
+**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
+
+# --hints--
+
+You should have a `h1` element with the text `Pricing Plans`
+
+```js
+const h1 = document.querySelector('h1');
+assert.exists(h1);
+assert.equal(h1.textContent, 'Pricing Plans', 'The h1 has the text Pricing Plans');
+```
+
+You should have three `div` elements with the class `pricing-card`
+
+```js
+const div = document.querySelectorAll('div.pricing-card');
+assert.exists(div);
+assert.lengthOf(div, 3, 'There are 3 div elements with the class pricing-card');
+```
+
+You should have a `div` element with the class `pricing-card` and `basic-plan`
+
+```js
+const div = document.querySelector('div.pricing-card.basic-plan');
+assert.exists(div);
+```
+
+You should have a `div` element with the class `pricing-card` and `premium-plan`
+
+```js
+const div = document.querySelector('div.pricing-card.premium-plan');
+assert.exists(div);
+```
+
+You should have a `div` element with the class `pricing-card` and `pro-plan`
+
+```js
+const div = document.querySelector('div.pricing-card.pro-plan');
+assert.exists(div);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>Pricing Plans Layout Page</title>
+  </head>
+  <body></body>
+</html>
+```
+
+```css
+
+```
+
+# --solutions--
```

Comment:
When I view the demo project, the layout seems off here. 

<img width="924" height="623" alt="Screenshot 2025-10-14 at 4 06 01 PM" src="https://github.com/user-attachments/assets/4325c654-1d8d-4aab-a676-c6c5f07daa27" />

We should change it to something like this down below for larger screens. and then smaller screens can still be in a column

<img width="922" height="559" alt="Screenshot 2025-10-14 at 4 08 04 PM" src="https://github.com/user-attachments/assets/29c0b956-793b-4a4f-9077-d9a5decf0799" />

Reply:
Working on this now